### ISSUE TITLE
x-mac-cyrillic UTF-8

### ISSUE THREAD

**Issue by Kuzuto** [php-mime-mail-parser/php-mime-mail-parser#230]
Hi,
Got the iconv(): Wrong charset on the charset "x-mac-cyrillic".
I changed:
'x-mac-cyrillic' => 'x-mac-cyrillic',
to:
'x-mac-cyrillic' => 'UTF-8',
and then I could getMessageBody() without problem.

Is that possible to do ?

**Comment by eXorus** [comment]
Hi,

Could you provide me an email with the result expected ? or only one header encoded in x-mac-cyrillic.

Something like that:
```
Subject: =?windows-1250?Q?Automatyczna_odpowied=9F:_Piotrze,_test_z_6_miesi=EAcy_na?=
 =?windows-1250?Q?uki_ci=B9gle_na_Ciebie_czeka?=
```
 Subject expected:
`Automatyczna odpowiedź: Piotrze, test z 6 miesięcy nauki ciągle na Ciebie czeka`

Thanks for your report.

**Comment by Kuzuto** [comment]
I suspect something is wrong i the email.. I can't find any text with different encoding.. here is the full eml :
#####

Received: from news-s.info (unknown [213.91.210.186])
	by xxx.xxx.dk with ESMTPS
	id 4a799634-4ae4-11e9-aeda-005056a84308;
	Wed, 20 Mar 2019 07:46:37 +0000 (UTC)
dkim-signature: v=1; a=rsa-sha256; d=news-s.info; s=selector;
	c=relaxed/relaxed; q=dns/txt; h=From:Reply-To:Subject:Date:Message-ID:To:MIME-Version:Content-Type;
	bh=awfWZB4udiYLWm7GwYu8vCS/3QjnDTvOUDLi9AMTMZk=;
	b=UIujdBxsXuS/IBV9bHVnekm04kUCOH6B0vdks5T00KTU5mtUGPUn0HVhoL5pAM9+6/vLDaEdI8oLWHCoO3PicM/ZOclJ23JLksJw9ukkXsK1sFDMXrRapWBeM3QiK/kWOV7lLL904aDxpds2pKsqWV9VeKSoi4G7rMJDvmD19K1RM+OP71Un8Z+zkat7XetZSY36Zoe+yteDlLfaXQwZvr1p0JrBDTMYex+S/ryCEr84QvyotFKaSlmRpX
	joOLa9HaSqyor4+I12PlG4h0bKIeHsUU2C81JIJQ1EMVBZhd8Dnab1MhSQTrku3DWKiZ2YN/mlhxJLHvrElBB5bAApYQ==
Received: from DESKTOP57VP31H (news-s.info [213.91.210.186])
	by news-s.info with ESMTPA
	; Wed, 20 Mar 2019 09:46:38 +0200
Return-Path: science@news-s.info
MIME-Version: 1.0
From: "Science and Education Foundation, Bulgaria" <science@news-s.info>
Reply-To: science@news-s.info
To: xxx@xxx.dk
Subject: Scientific conferences, Bulgaria 2019
Content-Type: multipart/alternative;
	boundary="----=_NextPart_001_0255_05604769.13815D8D"
Date: Wed, 20 Mar 2019 09:46:38 +0200
Message-ID: <741248069714414411836@DESKTOP-57VP31H>


------=_NextPart_001_0255_05604769.13815D8D
Content-Type: text/plain; charset="x-mac-cyrillic"
Content-Transfer-Encoding: quoted-printable

ACADEMIC NEWSLETTER #11 | March 2019
----------------------------------------------------------
View email in browser: http://scientific-news.net/ise/ise19-eng.html


Dear Colleague,

We are pleased to inviteyou to participate in conferences, part of Internat=
ional Scientific Events2019, to be held in Flora Expo Center, Burgas on the=
 Bulgarian Black Sea Coast(EU).

The conferences areorganized by the Bulgarian Academy of Sciences, the Unio=
n of Scientists inBulgaria, the Science and Education Foundation and partne=
rs.

Agriculture & Food, 7th International Conference (25-29 June)
http://sciencebulgaria.org/agriculture

Ecology & Safety, 28th International Conference (28 June - 2 July)
http://sciencebulgaria.org/ecology

Materials, Methods & Technologies, 21st International Conference (1-5 July)
http://sciencebulgaria.org/materials

Economy & Business, 18th International Conference (20-24 August)
http://sciencebulgaria.org/economy

Education, Research & Development, 10th International Conference (23-27 Aug=
ust)
http://sciencebulgaria.org/education

Language, Individual & Society, 13th International Conference (26-30 August)
http://sciencebulgaria.org/language

The papers presented will be published in open-access journals, part of Int=
ernational Scientific Publications (http://sciencebulgaria.org/journals-en).

Should you require further information, please don't hesitate to contactus:=
 http://bit.ly/2lEe8Aa

Tell a colleague about our conferences. Even just share this email!

Sincerely,

The ISE Team

-----------------------------------------------

The Science and Education Foundation only usesyour e-mail address to inform=
 you of up-to-date scientific events andprojects.

Not interested=3F To stop receiving further communication from us, visit ht=
tp://sciencebulgaria.org/unsubscribe

=C2=A9 2019 Science and Education Foundation, Bulgaria. All Rights Reserved.

------=_NextPart_001_0255_05604769.13815D8D
Content-Type: text/html; charset="x-mac-cyrillic"
Content-Transfer-Encoding: quoted-printable

<head>=0A            =0A<meta http-equiv=3D"Content-Type" content=3D"text/h=
tml; charset=3Dutf-8"> =0A<meta name=3D"GENERATOR" content=3D"MSHTML 11.00.=
10570.1001"></head> =0A<body>=0A<p><font face=3D"Arial, Helvetica, sans-ser=
if" size=3D"2"><strong><u>ACADEMIC       =0A NEWSLETTER #11&nbsp;|&nbsp;Mar=
ch 2019</u></strong></font></p>=0A<p><font face=3D"Arial, Helvetica, sans-s=
erif" size=3D"2"><font face=3D"Arial, Helvetica, sans-serif" size=3D"2"><a =
href=3D"http://scientific-news.net/ise/ise19-eng.html=3Futm_source=3DSendBl=
aster&utm_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu=
2%2D19%2E03">view email in =0A browser</a></font></font></p>=0A<p><font fac=
e=3D"Arial, Helvetica, sans-serif" size=3D"2"><font face=3D"Arial, Helvetic=
a, sans-serif" size=3D"2">Dear Colleague,</font></font></p>=0A<p><font face=
=3D"Arial, Helvetica, sans-serif" size=3D"2">We are pleased to invite   =0A=
 you to participate in conferences, part of International Scientific Events=
      =0A 2019,  to be held in <a href=3D"http://bit.ly/2Khzt0m=3Futm_sourc=
e=3DSendBlaster&utm_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_cam=
paign=3DEu2%2D19%2E03">Flora Expo Center, Burgas =0A</a>on the Bulgarian Bl=
ack Sea Coast   (EU).</font></p>=0A<p><font face=3D"Arial, Helvetica, sans-=
serif" size=3D"2">The conferences are       =0A organized by the Bulgarian =
Academy of Sciences, the Union of Scientists in      =0A  Bulgaria, the Sci=
ence and Education Foundation and partners.<font face=3D"Arial, Helvetica, =
sans-serif" size=3D"2"><br></font></font></p>=0A<p><font face=3D"Arial, Hel=
vetica, sans-serif" size=3D"2"><strong>Agriculture &amp;  =0A Food</strong>=
, 7th International Conference (25-29 June) <a href=3D"http://sciencebulgar=
ia.org/agriculture=3Futm_source=3DSendBlaster&utm_medium=3Demail&utm_term=
=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2D19%2E03">Read =0A more</a></f=
ont></p>=0A<p><font face=3D"Arial, Helvetica, sans-serif" size=3D"2"><stron=
g>Ecology &amp;      =0A Safety</strong>, 28th International Conference (28=
 June - 2 July) <a href=3D"http://sciencebulgaria.org/ecology=3Futm_source=
=3DSendBlaster&utm_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_camp=
aign=3DEu2%2D19%2E03">Read =0A more</a></font></p>=0A<p><font face=3D"Arial=
, Helvetica, sans-serif" size=3D"2"><strong>Materials, Methods =0A&amp; Tec=
hnologies</strong>, 21st International Conference (1-5 July) <a href=3D"htt=
p://sciencebulgaria.org/materials=3Futm_source=3DSendBlaster&utm_medium=3De=
mail&utm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2D19%2E03">Read =
=0A more</a></font></p>=0A<p><font face=3D"Arial, Helvetica, sans-serif" si=
ze=3D"2"><font face=3D"Arial, Helvetica, sans-serif" size=3D"2"><strong>Eco=
nomy &amp;  Business</strong>, 18th International Conference =0A (20-24 Aug=
ust) <a href=3D"http://sciencebulgaria.org/economy=3Futm_source=3DSendBlast=
er&utm_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2=
D19%2E03">Read      =0Amore</a></font></font></p>=0A<p><font face=3D"Arial,=
 Helvetica, sans-serif" size=3D"2"><strong>Education,        =0AResearch &a=
mp; Development</strong>, 10th International Conference (23-27 =0AAugust)  =
             <a href=3D"http://sciencebulgaria.org/education=3Futm_source=
=3DSendBlaster&utm_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_camp=
aign=3DEu2%2D19%2E03">Read   =0A more</a></font></p>=0A<p><font face=3D"Ari=
al, Helvetica, sans-serif" size=3D"2"><strong>Language,        =0AIndividua=
l &amp; Society</strong>, 13th International Conference (26-30 August) =0A<=
a href=3D"http://sciencebulgaria.org/language=3Futm_source=3DSendBlaster&ut=
m_medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2D19%2=
E03">Read  more</a></font></p><font face=3D"Arial, Helvetica, sans-serif" s=
ize=3D"2"><font face=3D"Arial, Helvetica, sans-serif" size=3D"2"><font face=
=3D"Arial, Helvetica, sans-serif" size=3D"2">=0A<p><br>The papers presented=
  will be published in open-access journals, part of  =0A            <a hre=
f=3D"http://sciencebulgaria.org/journals-en=3Futm_source=3DSendBlaster&utm_=
medium=3Demail&utm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2D19%2E0=
3">International    =0A Scientific   Publications</a>.</p>=0A<p>Should you =
require further information, please don't hesitate to <a href=3D"http://bit=
.ly/2lEe8Aa=3Futm_source=3DSendBlaster&utm_medium=3Demail&utm_term=3D12&utm=
_content=3Dnews&utm_campaign=3DEu2%2D19%2E03">contact =0A us</a>.</p>=0A<p>=
Tell a colleague about our conferences. Even just share this email!</p>=0A<=
p>Sincerely,</p>=0A<p>The ISE Team<font face=3D"Arial, Helvetica, sans-seri=
f" size=3D"2"><br></font></p></font></font></font> =0A<p><font face=3D"Aria=
l" size=3D"2">________________________</font></p>=0A<p><font face=3D"Arial"=
 size=3D"1">The Science and Education Foundation only uses    =0A your e-ma=
il address to inform you of up-to-date scientific events and       =0A proj=
ects.</font></p>=0A<p><font face=3D"Arial"><font size=3D"1"><font face=3D"A=
rial, Helvetica, sans-serif">Not interested=3F Click <a href=3D"http://scie=
ncebulgaria.org/unsubscribe=3Futm_source=3DSendBlaster&utm_medium=3Demail&u=
tm_term=3D12&utm_content=3Dnews&utm_campaign=3DEu2%2D19%2E03">here</a> =0At=
o stop receiving further  communication from us.</font></font></font></p>=
=0A<p><font face=3D"Arial" size=3D"2"><font size=3D"1">=C2=A9 2019 Science =
and Education       =0A Foundation, Bulgaria. All Rights Reserved.</font><b=
r></font></p>=0A</body>
------=_NextPart_001_0255_05604769.13815D8D--

**Comment by eXorus** [comment]
Fixed in 4.0.3

I did another solution but it should work for you.
